css/src/lib.rs

```diff
@@ -109,22 +105,19 @@ const RESERVED3_LEN: usize = 32;
 const RESERVED4_LEN: usize = 12;
 
 /// The SIGSTRUCT structure
-#[repr(C, packed)]
-#[derive(Clone)]
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct Signature {
     /// The byte stream 0x06000000E10000000000010000000000
     header: [u8; HEADER_LEN],
     /// Intel enclaves will use 0x00008086, other enclaves will use 0x00000000
-    vendor: [u8; VENDOR_LEN],
+    vendor: EnclaveVendor,
```

Comment:
Since this is no longer repr C and nom is unpacking this can store the rust types directly.
Doing this ensures that we only fail on load and once the struct is loaded it's guaranteed in a good state.

I would like to do the same for the other types, but that would require bringing in some of the `mc-sgx-***-sys-types` crates as dependencies

Reply:
Yeah, we can discuss. The main thing that prevented me from doing that in the past was the linkage in `sgx_types`, but that's not really an issue in the new-style crates.